### PR TITLE
Time shift material folder end_at time when duplicating.

### LINKED ISSUE #1771: Time shift material folder end_at time when duplicating
If an `end_at` time is specified, it should also be time shifted by the same amount as the course.

### PR DESCRIPTION
**Pull request by fonglh**
Add spec to test that the start/end times are shifted.

Fixes #1771.

### CODE REVIEW

File: spec/services/course/duplication_service_spec.rb [hunk 1]
```diff
@@ -217,6 +217,21 @@
...
+          # Select just the folders that were created in this context and sort them by name.
+          new_standalone_folders = @new_folders.select do |folder|
+            folders.map(&:name).include?(folder.name)
+          end
+          new_standalone_folders.sort_by!(&:name)
+
+          folders_array = new_standalone_folders.zip(folders)
+          folders_array.each do |new_folder, original_folder|
+            # 1.day is the time shift specified for this duplication spec
+            expect(new_folder.start_at).to be_within(10.minutes).of original_folder.start_at + 1.day
```
Comment (allenwq) [inline spec/services/course/duplication_service_spec.rb]: Is `10min` too big ?  the time should be exact `start + shift` right ?

Comment (fonglh) [inline spec/services/course/duplication_service_spec.rb]: I did try 1 minute, which passed locally but not on travis.

Comment (fonglh) [inline spec/services/course/duplication_service_spec.rb]:
```expect(new_folder.start_at).to be_within(1.minute).of original_folder.start_at + 1.day
       expected 2017-01-02 23:14:53.973377000 +0000 to be within 60 of 2017-01-02 23:12:45 UTC```

There's a 2+ minute gap.

**Comment by fonglh** [comment]
On travis, the times are sometimes quite off.
```
expect(new_folder.start_at).to be_within(5.minutes).of original_folder.start_at + 1.day
       expected 2017-01-02 23:59:15.970666000 +0000 to be within 300 of 2017-01-02 23:51:46 UTC
```

This one is 8 minutes off.

**Comment by weiqingtoh** [comment]
Possible to check how much it differs in local, and does it happen on both ruby versions? 

I've not had such a large differential before though, not sure what is happening.

**Comment by allenwq** [comment]
@fonglh Assessment start_at will overwrite the start_at of the owner folder, check that also ?

**Comment by fonglh** [comment]
@weiqingtoh local has a 10 second difference.
```
 Failure/Error: expect(new_folder.start_at).to be_within(1.seconds).of original_folder.start_at + 1.day
       expected 2017-01-05 03:08:29.972911000 +0000 to be within 1 of 2017-01-05 03:08:19 UTC
```

It happens on travis on both local versions. A few minutes of difference.

**Comment by fonglh** [comment]
I think I know why what's causing the large time shift difference, will fix that first.

**Pull request merged**